Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/EMStudioSDK/Source/MotionEventPresetManager.cpp

```diff
@@ -302,25 +302,28 @@ namespace EMStudio
         }
     }
 
-
     // Check if motion event with this configuration exists and return color.
     AZ::u32 MotionEventPresetManager::GetEventColor(const EMotionFX::EventDataSet& eventDatas) const
     {
         for (const MotionEventPreset* preset : m_eventPresets)
         {
-            EMotionFX::EventDataSet commonDatas;
             const EMotionFX::EventDataSet& presetDatas = preset->GetEventDatas();
-            const bool allMatch = AZStd::all_of(presetDatas.cbegin(), presetDatas.cend(), [eventDatas](const EMotionFX::EventDataPtr& presetData)
+
+            const size_t numEventDatas = eventDatas.size();
+            if (numEventDatas == presetDatas.size())
             {
-                const auto thisPresetDataHasMatch = AZStd::find_if(eventDatas.cbegin(), eventDatas.cend(), [presetData](const EMotionFX::EventDataPtr& eventData)
+                for (size_t i = 0; i < numEventDatas; ++i)
                 {
-                    return ((presetData && eventData && *presetData == *eventData) || (!presetData && !eventData));
-                });
-                return thisPresetDataHasMatch != eventDatas.cend();
-            });
-            if (allMatch)
-            {
-                return preset->GetEventColor();
+                    const EMotionFX::EventDataPtr& eventData = eventDatas[i];
+                    const EMotionFX::EventDataPtr& presetData = presetDatas[i];
+
+                    if (eventData && presetData &&
+                        eventData->RTTI_GetType() == presetData->RTTI_GetType() &&
+                        *eventData == *presetData)
+                    {
```

Comment:
The `EventData` subclasses all have to implement `bool Equal(const EventData& rhs)`, which all internally are going to do that RTTI check. So we don't need it here, it is up to the EventData types themselves to determine if they are equal or not.
```suggestion
                    if (eventData && presetData && *eventData == *presetData)
                    {
```

Reply:
Ah great, thanks. I've replaced the manual checks with the comparison operator for the shared_ptr which calls the class operators in case they are not null.